js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/PartitionSubsetList.tsx

```diff
@@ -0,0 +1,193 @@
+          <SpinnerWithText label="Loading partition keys…" />
+        </Box>
+      );
+    }
+
+    return (
+      <>
+        {partitionKeys.length > MAX_ITEMS_BEFORE_TRUNCATION ? (
+          <SearchContainer padding={{vertical: 4, horizontal: 8}}>
+            <TextInput
+              icon="search"
+              placeholder="Filter partitions…"
+              value={searchValue}
+              onChange={(e) => setSearchValue(e.target.value)}
+            />
+          </SearchContainer>
+        ) : null}
+        <div
+          style={{
+            height: count > MAX_ITEMS_BEFORE_TRUNCATION ? '150px' : count * ITEM_HEIGHT + 16,
+            overflow: 'hidden',
+          }}
+        >
+          <Container ref={container}>
+            <Menu>
+              <Inner $totalHeight={totalHeight}>
+                {virtualItems.map(({index, key, size, start}) => {
+                  const partitionKey = filteredKeys[index]!;
+                  return (
+                    <Row $height={size} $start={start} key={key}>
+                      <MenuItem
+                        onClick={() => {
+                          selectPartition && selectPartition(partitionKey);
+                        }}
+                        text={
+                          <Box flex={{direction: 'row', alignItems: 'center', gap: 8}}>
+                            <PartitionStatusDot $color={color} $hoverColor={hoverColor} />
+                            <div>
+                              <MiddleTruncate text={partitionKey} />
+                            </div>
+                          </Box>
+                        }
+                      />
+                    </Row>
+                  );
+                })}
+              </Inner>
+            </Menu>
+          </Container>
+        </div>
+      </>
+    );
+  };
+
+  return (
+    <div style={{width: '292px'}}>
+      <Box
+        padding={{vertical: 8, left: 12, right: 8}}
+        border="bottom"
+        flex={{direction: 'row', alignItems: 'center', justifyContent: 'space-between'}}
+        style={{display: 'grid', gridTemplateColumns: 'minmax(0, 1fr) auto', gap: 8}}
+      >
+        <strong>
+          <MiddleTruncate text={description} />
+        </strong>
+        {status ? <PolicyEvaluationStatusTag status={status} /> : null}
+      </Box>
+      {content()}
+    </div>
+  );
+};
+
+const statusToColors = (status: AssetConditionEvaluationStatus) => {
+  switch (status) {
+    case AssetConditionEvaluationStatus.TRUE:
+      return {color: Colors.accentGreen(), hoverColor: Colors.accentGreenHover()};
+    case AssetConditionEvaluationStatus.FALSE:
+      return {color: Colors.accentYellow(), hoverColor: Colors.accentYellowHover()};
+    case AssetConditionEvaluationStatus.SKIPPED:
+      return {color: Colors.accentGray(), hoverColor: Colors.accentGrayHover()};
+    default:
+      return assertUnreachable(status);
+  }
+};
```

Comment:
This could also be a dictionary.

```
const statusToColors: Record<AssetConditionEvaluationStatus, {color: string, hoverColor: string}> = {
  [AssetConditionEvaluationStatus.TRUE]: {...},
  [AssetConditionEvaluationStatus.FALSE]: {...},
  [AssetConditionEvaluationStatus.SKIPPED]: {...},
}
```
This way we don't need an `assertUnreachable` because by using Record we're telling typescript this object has to contain all of the possible keys.